### PR TITLE
#38 - Add ProcDomainMemoryStats procedure. 

### PR DESCRIPTION
**Pull request by amandahla**
Todo:
- [x] Test

### CODE REVIEW

**Comment by amandahla** [comment]
@benlemasurier Sorry, I forgot golint!

File: libvirt.go [hunk 1]
```diff
@@ -318,6 +320,44 @@ const (
 	RebootParavirt
 )
 
+// DomainMemoryStatTag specifies domain memory tags
+type DomainMemoryStatTag uint32
+
+const (
+	// DomainMemoryStatTagSwapIn - The total amount of data read from swap space (in kB).
+	DomainMemoryStatTagSwapIn DomainMemoryStatTag = 0 << iota
```
Comment (benlemasurier) [inline libvirt.go]: `DomainMemoryStatTagSwapIn DomainMemoryStatTag = iota`

File: libvirt.go [hunk 2]
```diff
@@ -318,6 +320,44 @@ const (
...
+	// DomainMemoryStatTagSwapOut - The total amount of memory written out to swap space (in kB).
+	DomainMemoryStatTagSwapOut
```
Comment (benlemasurier) [inline libvirt.go]: nitpick: let's add a `\n` after each const definition here to match the rest of the file.

Comment (amandahla) [inline libvirt.go]: Sure!

File: libvirt.go (hunk 1 above)
Comment (amandahla) [inline libvirt.go]: Thanks, changed!

File: libvirttest/libvirt.go [hunk 3]
```diff
@@ -207,6 +207,26 @@ var testDomainsReply = []byte{
 	0x00, 0x00, 0x02,
 }
 
+var testDomainMemoryStatsReply = []byte{
+	0x00, 0x00, 0x00, 0x38, // length
+	0x20, 0x00, 0x80, 0x86, // program
+	0x00, 0x00, 0x00, 0x01, // version
+	0x00, 0x00, 0x00, 0x9f, // procedure
+	0x00, 0x00, 0x00, 0x01, // type
+	0x00, 0x00, 0x00, 0x07, // serial
```
Comment (benlemasurier) [inline libvirttest/libvirt.go]: `0x00, 0x00, 0x00, 0x00`

sorry, not obvious but the serial is managed automatically.

Comment (amandahla) [inline libvirttest/libvirt.go]: I didn't notice. Changed now!

**Comment by benlemasurier** [comment]
Thanks @amandahla! The PR is still titled as WIP, are you still planning additional changes?

**Comment by amandahla** [comment]
@benlemasurier I was planning to document how I did to find out the call/reply from the procedure but I can do it later. What do you think?

**Comment by benlemasurier** [comment]
@amandahla writing that up would be extremely useful for a lot of folks. I think another PR makes the most sense. I'll go ahead and merge this one for now. Thanks again for your contribution.

**Pull request merged**